logstash-core/src/main/java/org/logstash/plugins/codecs/Debug.java

```diff
@@ -0,0 +1,96 @@
+package org.logstash.plugins.codecs;
+
+import co.elastic.logstash.api.Codec;
+import co.elastic.logstash.api.Configuration;
+import co.elastic.logstash.api.Context;
+import co.elastic.logstash.api.Event;
+import co.elastic.logstash.api.LogstashPlugin;
+import co.elastic.logstash.api.PluginConfigSpec;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static org.logstash.ObjectMappers.JSON_MAPPER;
+
+/**
+ * An encode-only codec that prints events as pretty-printed JSON somewhat similar
+ * to the "rubydebug" codec.
+ */
+@LogstashPlugin(name = "debug")
+public class Debug implements Codec {
+
+    private static final PluginConfigSpec<Boolean> METADATA_CONFIG =
+            PluginConfigSpec.booleanSetting("metadata", false);
+
+    private final Context context;
+    private final boolean metadata;
+    private final String id;
+
+    private Function<Event, Map<String, Object>> dataFunction;
+
+    /**
+     * Required constructor.
+     *
+     * @param configuration Logstash Configuration
+     * @param context       Logstash Context
+     */
+    public Debug(final Configuration configuration, final Context context) {
+        this(context, configuration.get(METADATA_CONFIG));
+    }
+
+    private Debug(Context context, boolean metadata) {
+        this.context = context;
+        this.metadata = metadata;
+        this.dataFunction = this.metadata ? this::withMetadata : this::withoutMetadata;
+        this.id = UUID.randomUUID().toString();
+    }
+
+    @Override
+    public void decode(ByteBuffer buffer, Consumer<Map<String, Object>> eventConsumer) {
+        throw new UnsupportedOperationException("The debug codec supports only encoding");
+    }
+
+    @Override
+    public void flush(ByteBuffer buffer, Consumer<Map<String, Object>> eventConsumer) {
+    }
+
+    @Override
+    public void encode(Event event, OutputStream output) throws IOException {
+        String formatted = JSON_MAPPER
+                .writerWithDefaultPrettyPrinter()
+                .writeValueAsString(dataFunction.apply(event)) + System.lineSeparator();
```

Comment:
why the `String` intermediary? I think we can use `ObjectMapper#writeValueAsBytes(Object)`.